platform-service/src/unstract/platform_service/main.py

```diff
@@ -200,33 +228,63 @@ def usage() -> Any:
         input_tokens=input_tokens,
         output_tokens=completion_tokens,
     )
-
-    query = f"""
-        INSERT INTO "{org_id}"."token_usage" (id, workflow_id,
-        execution_id, adapter_instance_id, run_id, usage_type,
-        llm_usage_reason, model_name, embedding_tokens, prompt_tokens,
-        completion_tokens, total_tokens, cost_in_dollars, created_at, modified_at)
-        VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
-    """
     usage_id = uuid.uuid4()
     current_time = datetime.now()
-    params = (
-        usage_id,
-        workflow_id,
-        execution_id,
-        adapter_instance_id,
-        run_id,
-        usage_type,
-        llm_usage_reason,
-        model_name,
-        embedding_tokens,
-        prompt_tokens,
-        completion_tokens,
-        total_tokens,
-        cost_in_dollars,
-        current_time,
-        current_time,
-    )
+    if check_feature_flag_status(FeatureFlag.MULTI_TENANCY_V2):
+        query = f"""
+            INSERT INTO {DBTableV2.TOKEN_USAGE} (id, organization_id, workflow_id,
+            execution_id, adapter_instance_id, run_id, usage_type,
+            llm_usage_reason, model_name, embedding_tokens, prompt_tokens,
+            completion_tokens, total_tokens, cost_in_dollars, created_at, modified_at)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+        """
+        usage_id = uuid.uuid4()
+        current_time = datetime.now()
+        params = (
+            usage_id,
+            organization_uid,
+            workflow_id,
+            execution_id,
+            adapter_instance_id,
+            run_id,
+            usage_type,
+            llm_usage_reason,
+            model_name,
+            embedding_tokens,
+            prompt_tokens,
+            completion_tokens,
+            total_tokens,
+            cost_in_dollars,
+            current_time,
+            current_time,
+        )
+    else:
+        query = f"""
+            INSERT INTO "{org_id}"."token_usage" (id, workflow_id,
+            execution_id, adapter_instance_id, run_id, usage_type,
+            llm_usage_reason, model_name, embedding_tokens, prompt_tokens,
+            completion_tokens, total_tokens, cost_in_dollars, created_at, modified_at)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+        """
+        usage_id = uuid.uuid4()
+        current_time = datetime.now()
+        params = (
+            usage_id,
+            workflow_id,
+            execution_id,
+            adapter_instance_id,
+            run_id,
+            usage_type,
+            llm_usage_reason,
+            model_name,
+            embedding_tokens,
+            prompt_tokens,
+            completion_tokens,
+            total_tokens,
+            cost_in_dollars,
+            current_time,
+            current_time,
+        )
```

Comment:
@muhammad-ali-e same here, can we use SQLAlchemy here ?

Reply:
Addressed in above commend